homeassistant/components/onvif/camera.py

```diff
@@ -93,59 +70,37 @@
     }
 )
 
-SERVICE_PTZ_SCHEMA = vol.Schema(
-    {
-        ATTR_ENTITY_ID: cv.entity_ids,
-        vol.Optional(ATTR_PAN): vol.In([DIR_LEFT, DIR_RIGHT]),
-        vol.Optional(ATTR_TILT): vol.In([DIR_UP, DIR_DOWN]),
-        vol.Optional(ATTR_ZOOM): vol.In([ZOOM_OUT, ZOOM_IN]),
-        ATTR_MOVE_MODE: vol.In([CONTINUOUS_MOVE, RELATIVE_MOVE, ABSOLUTE_MOVE]),
-        vol.Optional(ATTR_CONTINUOUS_DURATION, default=0.5): cv.small_float,
-        vol.Optional(ATTR_DISTANCE, default=0.1): cv.small_float,
-        vol.Optional(ATTR_SPEED, default=0.5): cv.small_float,
-    }
-)
-
 
 async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
     """Set up a ONVIF camera."""
-    _LOGGER.debug("Setting up the ONVIF camera platform")
-
-    async def async_handle_ptz(service):
-        """Handle PTZ service call."""
-        pan = service.data.get(ATTR_PAN)
-        tilt = service.data.get(ATTR_TILT)
-        zoom = service.data.get(ATTR_ZOOM)
-        distance = service.data[ATTR_DISTANCE]
-        speed = service.data[ATTR_SPEED]
-        move_mode = service.data.get(ATTR_MOVE_MODE)
-        continuous_duration = service.data[ATTR_CONTINUOUS_DURATION]
-        all_cameras = hass.data[ONVIF_DATA][ENTITIES]
-        entity_ids = await async_extract_entity_ids(hass, service)
-        target_cameras = []
-        if not entity_ids:
-            target_cameras = all_cameras
-        else:
-            target_cameras = [
-                camera for camera in all_cameras if camera.entity_id in entity_ids
-            ]
-        for camera in target_cameras:
-            await camera.async_perform_ptz(
-                pan, tilt, zoom, distance, speed, move_mode, continuous_duration
-            )
-
-    hass.services.async_register(
-        DOMAIN, SERVICE_PTZ, async_handle_ptz, schema=SERVICE_PTZ_SCHEMA
-    )
-
-    _LOGGER.debug("Constructing the ONVIFHassCamera")
-
-    hass_camera = ONVIFHassCamera(hass, config)
+    LOGGER.warning("Setting up the ONVIF camera via configuration.yaml is deprecated")
+    return True
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Axis camera video stream."""
+    config = {
+        CONF_NAME: config_entry.data[CONF_NAME],
+        CONF_HOST: config_entry.data[CONF_HOST],
+        CONF_PORT: config_entry.data[CONF_PORT],
+        CONF_USERNAME: config_entry.data[CONF_USERNAME],
+        CONF_PASSWORD: config_entry.data[CONF_PASSWORD],
+        CONF_EXTRA_ARGUMENTS: config_entry.options[CONF_EXTRA_ARGUMENTS],
+        CONF_RTSP_TRANSPORT: config_entry.options[CONF_RTSP_TRANSPORT],
+    }
 
-    await hass_camera.async_initialize()
+    entities = []
+    active_profiles = [
+        key for key, value in config_entry.data[CONF_PROFILE].items() if value
+    ]
+    for profile in active_profiles:
```

Comment:
Why create an entity per profile? Isn't profile things like different resolutions?

Reply:
For single camera devices, yes. For NVRs profiles are different cameras.

Reply:
Based on conversation on Discord, we went with adding all profiles automatically, but putting the profile name in the entity name so people can easily disable the ones they don't want/need.